src/main/webapp/app/exam/manage/students/room-distribution/students-room-distribution.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable, Signal, WritableSignal, computed, inject, signal } from '@angular/core';
+import { HttpClient, HttpParams, HttpResponse } from '@angular/common/http';
+import { Observable, catchError, map, throwError } from 'rxjs';
+import { ExamDistributionCapacityDTO, RoomForDistributionDTO } from 'app/exam/manage/students/room-distribution/students-room-distribution.model';
+
+@Injectable({ providedIn: 'root' })
+export class StudentsRoomDistributionService {
+    private http = inject(HttpClient);
+
+    private readonly BASE_URL = 'api/exam';
+
+    private availableRoomsInternal: WritableSignal<RoomForDistributionDTO[] | undefined> = signal(undefined);
+    private capacityDataInternal: WritableSignal<ExamDistributionCapacityDTO | undefined> = signal(undefined);
+
+    // Signals that can be used by other components
+    readonly availableRooms: Signal<RoomForDistributionDTO[]> = computed(() => this.availableRoomsInternal() ?? []);
+    readonly capacityData: Signal<ExamDistributionCapacityDTO> = computed(
+        () =>
+            this.capacityDataInternal() ??
+            ({
+                combinedDefaultCapacity: 0,
+                combinedMaximumCapacity: 0,
+            } as ExamDistributionCapacityDTO),
+    );
```

Comment:
I do not understand that comment, is that a reasoning why they are not private? What does the comment intend to say? 😅 

```suggestion
    readonly availableRooms: Signal<RoomForDistributionDTO[]> = computed(() => this.availableRoomsInternal() ?? []);
    readonly capacityData: Signal<ExamDistributionCapacityDTO> = computed(
        () =>
            this.capacityDataInternal() ??
            ({
                combinedDefaultCapacity: 0,
                combinedMaximumCapacity: 0,
            } as ExamDistributionCapacityDTO),
    );
```